src/jdk.management/share/native/libmanagement_ext/DiagnosticCommandImpl.c

```diff
@@ -173,10 +181,17 @@ Java_com_sun_management_internal_DiagnosticCommandImpl_getDiagnosticCommandInfo
   }
   jmm_interface->GetDiagnosticCommandInfo(env, commands, dcmd_info_array);
   for (i=0; i<num_commands; i++) {
+      // Ensure capacity for 6 + 3 local refs:
+      //  6 => jname, jdesc, jimpact, cmd, args, obj
+      //  3 => permission class, name, action
+      (*env)->PushLocalFrame(env, 6 + 3);
+
+      cmd = (*env)->GetObjectArrayElement(env, commands, i);
       args = getDiagnosticCommandArgumentInfoArray(env,
-                                                   (*env)->GetObjectArrayElement(env,commands,i),
+                                                   cmd,
                                                    dcmd_info_array[i].num_arguments);
       if (args == NULL) {
+          (*env)->PopLocalFrame(env, NULL);
```

Comment:
You actually need 2 PopLocalFrame calls here. One to match the PushLocalFrame inside this loop, and one to match the PushLocalFrame outside the loop.

Reply:
Fixed.